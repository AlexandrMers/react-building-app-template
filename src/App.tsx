import React, { FC, memo } from "react";

import styles from "./style.module.scss";

import background from "./example-img.jpg";

import Typography from "@/components/Typography";

const App: FC = (props) => {
  return (
    <div className={styles.App}>
      <h1 className="red">React App with custom webpack config</h1>
      <img src={background} alt="horses" />
      <Typography>Helloo))</Typography>
    </div>
  );
};

export default memo(App);
