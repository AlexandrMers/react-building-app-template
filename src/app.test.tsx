import React from "react";
import { render, screen } from "@testing-library/react";

// Components
import App from "@/App";

describe("src/App.tsx", () => {
  describe("не передали свойств", () => {
    it("-> snapshot", () => {
      const { container } = render(<App />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('-> отобразится заголовок "React App with custom webpack config"', () => {
      render(<App />);
      const title = screen.getByText(/React App with custom webpack config/i);
      expect(title).toBeInTheDocument();
    });
  });
});
