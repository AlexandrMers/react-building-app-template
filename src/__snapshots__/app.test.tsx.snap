// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/App.tsx не передали свойств -> snapshot 1`] = `
<div
  class="App"
>
  <h1
    class="red"
  >
    React App with custom webpack config
  </h1>
  <img
    alt="horses"
    src="[object Object]"
  />
  <h1>
    Helloo))
  </h1>
</div>
`;
